pkg/api/v1/token.go

```diff
@@ -32,25 +32,58 @@ func (g *TokenGroup) CreateWorkspaceToken(ctx echo.Context) error {
 	workspaceId := ctx.Param("workspaceId")
 	workspace, err := g.backendRepo.GetWorkspaceByExternalId(ctx.Request().Context(), workspaceId)
 	if err != nil {
-		return echo.NewHTTPError(http.StatusBadRequest, "Invalid workspace")
+		return ctx.JSON(http.StatusBadRequest, map[string]interface{}{
+			"error": "invalid workspace ID",
+		})
 	}
 
 	token, err := g.backendRepo.CreateToken(ctx.Request().Context(), workspace.Id, types.TokenTypeWorkspace, true)
 	if err != nil {
-		return echo.NewHTTPError(http.StatusInternalServerError, "Unable to create token")
+		return ctx.JSON(http.StatusInternalServerError, map[string]interface{}{
+			"error": "unable to create token",
+		})
 	}
 
 	return ctx.JSON(http.StatusOK, map[string]interface{}{
 		"token": token.Key,
 	})
 }
 
-type RevokeTokenRequest struct {
-	Token string `json:"token"`
+type ClusterAdminTokensRequestSerializer struct {
+	Disabled bool `json:"enabled"`
 }
 
-func (g *TokenGroup) RevokeWorkspaceToken(ctx echo.Context) error {
-	_, _ = ctx.(*auth.HttpAuthContext)
+func (g *TokenGroup) ClusterAdminUpdateAllWorkspaceTokens(ctx echo.Context) error {
```

Comment:
this should probably also be renamed - it doesn't just update tokens, it only disables them

Reply:
It toggles them